### PR TITLE
Feature/#20 liberty parser

### PR DESCRIPTION
**Pull request by TiagoAFontana**
This adds liberty parser and removes clang from travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -42,6 +36,6 @@ before_script:
   - cmake -DCMAKE_CXX_COMPILER=$COMPILER -DCMAKE_BUILD_TYPE=Release ..
 
 script:
-  - make -j
+  - make -j 1
```
Comment (csguth) [inline .travis.yml]: `make -j 1` is the same as just `make`, isn't it?

File: CMakeLists.txt [hunk 2]
```diff
@@ -24,18 +27,30 @@ find_package(Boost 1.59 REQUIRED)
 # Build LEF v5.8-p016 parser with the included Makefile
 include(ExternalProject)
 include_directories(.)
+ExternalProject_Add(
```
Comment (csguth) [inline CMakeLists.txt]: This is not necessary for this PR.

Comment (csguth) [reply CMakeLists.txt]: Nevermind.

File: CMakeLists.txt [hunk 3]
```diff
@@ -24,18 +27,30 @@ find_package(Boost 1.59 REQUIRED)
...
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND make installhdrs installlib
+        # LEF Makefile is not compatible with external builds
+        BUILD_IN_SOURCE 1
+)
 
+# Build Liberty parser
+find_package(GPERF REQUIRED)
 ExternalProject_Add(
-	lef_parser
```
Comment (csguth) [inline CMakeLists.txt]: Please, avoid swapping lines. This makes the diff harder to understand.

File: ophidian/parser/CMakeLists.txt [hunk 4]
```diff
@@ -9,13 +9,21 @@ set(PARSER_SOURCE
     ${CMAKE_CURRENT_SOURCE_DIR}/Def.cpp)
 
 LINK_DIRECTORIES(${THIRD_PARTY_PATH}/DEF/lib/ ${THIRD_PARTY_PATH}/LEF/lib/)
+#LINK_DIRECTORIES(${THIRD_PARTY_BUILD_PATH}/libertyParser/lib/ )
```
Comment (csguth) [inline ophidian/parser/CMakeLists.txt]: Commented lines are useless. Please, remove.

File: ophidian/parser/LibertyParser.cpp [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+#include <fstream>
+#include <cassert>
+#include <iostream>
+#include <sstream>
+
+namespace ophidian
+{
+namespace parser
+{
+
```
Comment (csguth) [inline ophidian/parser/LibertyParser.cpp]: Blank lines.

File: ophidian/parser/LibertyParser.cpp [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+LibertyParser::~LibertyParser()
+{
+
+}
+std::vector<double> LibertyParser::Pimpl::split_string_into_values(const std::string &the_string) {
+    std::vector<double> values;
+    std::stringstream input(the_string);
+    std::string segment;
+    while (std::getline(input, segment, ','))
+        values.push_back(std::stod(segment));
```
Comment (csguth) [inline ophidian/parser/LibertyParser.cpp]: Please, enclose with brackets.
```C++
while(cond)
{
    ...
}

File: ophidian/parser/LibertyParser.cpp [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+    return values;
+}
+
+void LibertyParser::Pimpl::read_LUT(si2drGroupIdT group, Liberty::LUT & lut) {
+    si2drAttrsIdT attrs = si2drGroupGetAttrs(group, &err);
+    si2drAttrIdT attr;
+
+//    std::vector< double > loadValues(8);
```
Comment (csguth) [inline ophidian/parser/LibertyParser.cpp]: Please, remove commented lines.

File: test/input_files/input_files [hunk 8]
```diff
@@ -0,0 +1 @@
+/home/tiago/devel/ophidian/test/input_files
```
Comment (csguth) [inline test/input_files/input_files:1]: Please remove this symlink.

**Review by csguth — CHANGES_REQUESTED** [review]

**Comment by csguth** [comment]
Please, create a PR just for the travis change.

File: test/input_files/input_files (hunk 8 above)
Comment (TiagoAFontana) [reply test/input_files/input_files:1]: Ok. Removed. :)

**Pull request closed without merging**